src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/IndexItem.java

```diff
@@ -89,6 +89,73 @@ public enum Category {
         TAGS
     }
 
+    /**
+     * The kind of index item. Index items can represent program elements,
+     * various doc comment tags, or other resources such as summary pages.
+     *
+     * Values in this enum must match elements in array itemDesc in
+     * JavaScript file search.js.template!
+     */
+    public enum Kind {
+        // Members
+        ENUM_CONSTANT,
+        FIELD,
+        STATIC_FIELD,
+        CONSTRUCTOR,
+        ANNOTATION_ELEMENT,
+        METHOD,
+        STATIC_METHOD,
+        RECORD_COMPONENT,
+        // Types
+        ANNOTATION_TYPE,
+        ENUM,
+        INTERFACE,
+        RECORD,
+        CLASS,
+        EXCEPTION_CLASS,
+        // Tags
+        SEARCH_ITEM,
+        SYSTEM_PROPERTY,
+        SECTION,
+        EXTERNAL_SPEC,
+        // Other
+        SUMMARY_PAGE;
+
+        /**
+         * {@return the kind of index item for a program element}
+         */
+        public static Kind ofElement (Element elem, Utils utils) {
```

Comment:
```suggestion
        public static Kind ofElement(Element elem, Utils utils) {
```

Reply:
Thanks, @liach! I pushed the changes you suggested along with two small fixes.